L1Trigger/L1THGCal/src/be_algorithms/HGCalClusteringImpl.cc

```diff
@@ -270,3 +276,162 @@ void HGCalClusteringImpl::clusterizeNN( const edm::PtrVector<l1t::HGCalTriggerCe
 
 }
 
+
+
+/*** FW-algorithms ***/
+void HGCalClusteringImpl::clusterizeDRNN( const edm::PtrVector<l1t::HGCalTriggerCell> & triggerCellsPtrs, 
+                                        l1t::HGCalClusterBxCollection & clusters,
+                                        const HGCalTriggerGeometryBase & triggerGeometry
+    ){
+
+    bool isSeed[triggerCellsPtrs.size()];
+    std::vector<unsigned> seedPositions;
+    seedPositions.reserve( triggerCellsPtrs.size() );
+
+    /* search for cluster seeds */
+    int itc(0);
+    for( edm::PtrVector<l1t::HGCalTriggerCell>::const_iterator tc = triggerCellsPtrs.begin(); tc != triggerCellsPtrs.end(); ++tc,++itc ){
+        
+        double seedThreshold = ((*tc)->subdetId()==HGCHEB ? scintillatorSeedThreshold_ : siliconSeedThreshold_);
+
+        /* decide if is a seed, if yes store the position into of triggerCellsPtrs */
+        isSeed[itc]       = ( (*tc)->mipPt() > seedThreshold) ? true : false;
+        if( isSeed[itc] ) seedPositions.push_back( itc );
+
+        /* remove tc from the seed vector if is a NN of an other seed*/
+        for( auto pos : seedPositions ){
```

Comment:
I suppose that this for loop and NN check should be done only if `isSeed[itc]` is true